### PR TITLE
Arc 1438 Create tool to automatically fill descriptions from OpenAI and backfill descriptions

### PR DESCRIPTION
**Pull request by ryantimjohn**
This PR creates a tool in the notebook fill_in_descriptions_using_openai.ipynb to fill in missing descriptions on documentation. See notebook here:
https://github.com/bsd/dbt-arc-functions/blob/180ec2ed9c69eb7d40f41ea108d7988eeee83fca/utils/fill_in_descriptions_using_openai.ipynb

It then runs that notebook and fills in missing descriptions on documentation.

There's a ton of documentation that needed touching up so this PR is massive.

### CODE REVIEW

**Review by Frydafly — CHANGES_REQUESTED** [review]
woah! ok, this fills in all our documentation, alright. 

I definitely want to merge in the documentation changes, but I think it would be helpful to make some additions to the notebook and also our readme so that: 
- a new user knows when/if/how to use the notebook (in terms of steps, you would first create macros, then run create_standard_models in your client project, then use the codegen script to create documentation yaml structure and files... and then at that point you can run the open ai script to fill in the descriptions?)
- there are some variables in the notebook, like file_path that would be nice to have up top as constants where the user can fill in the appropriate file path for new models so that the notebook fills it in 
- its assumed that the user will have an OPEN AI key in their env vars-- maybe a URL to the open AI website + a note in the notebook to `export OPEN_AI_KEY=[your key]` before running the notebook?

**Comment by ryantimjohn** [comment]
@Frydafly, this all sounds reasonable! I suggest we pull out the note book from the PR so I can work on that next week and just make the docs changes.

File: pull_request_template.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 - [ ] Create branch of dbt-arc-functions using JIRA extension so that naming convention is consistent
 
-- [ ] If you've created a macro, make sure it is documented by running the notebook: `create_docs.ipynb` util in this repo
+- [ ] If you've created a macro, make sure it is documented by running the notebook: `create_docs.ipynb` util in this repo. From there, you'll have to either fill in descriptions by hand OR use our `fill_in_descriptions_using_openai.ipynb` to fill in descriptions using OpenAI's ChatGPT. Always make sure to check over ChatGPT's column descriptions, as they're likely not perfect.
```
Comment (ryantimjohn) [inline pull_request_template.md:5]: @Frydafly how does this look?

**Comment by ryantimjohn** [comment]
@Frydafly, actually, I was able to make the changes you requested I believe. Commented on the PR template, and here's the updated notebook:
https://github.com/bsd/dbt-arc-functions/blob/022fea9289032be482e8c08af9de96e6a440fc33/utils/fill_in_descriptions_using_openai.ipynb

**Comment by Frydafly** [comment]
Sounds good!

All the best,
Fryda
(she/her)

**Review by Frydafly — APPROVED** [review]
🥳

**Pull request merged**